From 3b7ba986b711981a5cc10f81a22c86f9b2aea65c Mon Sep 17 00:00:00 2001
From: John Villalovos <john.l.villalovos@intel.com>
Date: Tue, 17 Dec 2013 11:33:47 -0800
Subject: Add ability for project to be 'prebuilt'

Add support for the make target: PREBUILT-PROJECT-libmix

Support added for:
 * asfparser
 * imagedecoder
 * mix
 * mix_vbp/viddec_fw/fw/codecs/h264/parser
 * mix_vbp/viddec_fw/fw/codecs/mp4/parser
 * mix_vbp/viddec_fw/fw/codecs/vc1/parser
 * mix_vbp/viddec_fw/fw/codecs/vp8/parser
 * mix_vbp/viddec_fw/fw/parser
 * videodecoder
 * videoencoder

Issue: ADEV-437

Change-Id: I21afc3ca98c18f95fc01ab53d9e224f629d0cb47
Signed-off-by: John L. Villalovos <john.l.villalovos@intel.com>
---
 asfparser/Android.mk                               |    3 +++
 imagedecoder/Android.mk                            |    2 ++
 mix_vbp/viddec_fw/fw/codecs/h264/parser/Android.mk |    3 +++
 mix_vbp/viddec_fw/fw/codecs/mp4/parser/Android.mk  |    3 +++
 mix_vbp/viddec_fw/fw/codecs/vc1/parser/Android.mk  |    3 +++
 mix_vbp/viddec_fw/fw/codecs/vp8/parser/Android.mk  |    3 +++
 mix_vbp/viddec_fw/fw/parser/Android.mk             |    3 +++
 videodecoder/Android.mk                            |    3 +++
 videoencoder/Android.mk                            |    3 +++
 9 files changed, 26 insertions(+), 0 deletions(-)

diff --git a/asfparser/Android.mk b/asfparser/Android.mk
index 3fecfbf..45585f2 100644
--- a/asfparser/Android.mk
+++ b/asfparser/Android.mk
@@ -23,3 +23,6 @@ LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libasfparser
 
 include $(BUILD_SHARED_LIBRARY)
+
+PREBUILT_PROJECT := libmix
+include $(BUILD_PREBUILT_BUNDLE_CREATE)
diff --git a/imagedecoder/Android.mk b/imagedecoder/Android.mk
index b5d3c95..ad138ac 100644
--- a/imagedecoder/Android.mk
+++ b/imagedecoder/Android.mk
@@ -106,3 +106,5 @@ LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_SHARED_LIBRARY)
 
+PREBUILT_PROJECT := libmix
+include $(BUILD_PREBUILT_BUNDLE_CREATE)
diff --git a/mix_vbp/viddec_fw/fw/codecs/h264/parser/Android.mk b/mix_vbp/viddec_fw/fw/codecs/h264/parser/Android.mk
index 62fe53d..99f2474 100644
--- a/mix_vbp/viddec_fw/fw/codecs/h264/parser/Android.mk
+++ b/mix_vbp/viddec_fw/fw/codecs/h264/parser/Android.mk
@@ -62,4 +62,7 @@ LOCAL_SHARED_LIBRARIES := libmixvbp
 
 include $(BUILD_SHARED_LIBRARY)
 
+PREBUILT_PROJECT := libmix
+include $(BUILD_PREBUILT_BUNDLE_CREATE)
+
 endif
diff --git a/mix_vbp/viddec_fw/fw/codecs/mp4/parser/Android.mk b/mix_vbp/viddec_fw/fw/codecs/mp4/parser/Android.mk
index 012c8cd..722041a 100644
--- a/mix_vbp/viddec_fw/fw/codecs/mp4/parser/Android.mk
+++ b/mix_vbp/viddec_fw/fw/codecs/mp4/parser/Android.mk
@@ -27,3 +27,6 @@ LOCAL_SHARED_LIBRARIES :=		\
 	libmixvbp
 
 include $(BUILD_SHARED_LIBRARY)
+
+PREBUILT_PROJECT := libmix
+include $(BUILD_PREBUILT_BUNDLE_CREATE)
diff --git a/mix_vbp/viddec_fw/fw/codecs/vc1/parser/Android.mk b/mix_vbp/viddec_fw/fw/codecs/vc1/parser/Android.mk
index e2479fe..542cc49 100644
--- a/mix_vbp/viddec_fw/fw/codecs/vc1/parser/Android.mk
+++ b/mix_vbp/viddec_fw/fw/codecs/vc1/parser/Android.mk
@@ -35,3 +35,6 @@ LOCAL_SHARED_LIBRARIES := \
     libmixvbp
 
 include $(BUILD_SHARED_LIBRARY)
+
+PREBUILT_PROJECT := libmix
+include $(BUILD_PREBUILT_BUNDLE_CREATE)
diff --git a/mix_vbp/viddec_fw/fw/codecs/vp8/parser/Android.mk b/mix_vbp/viddec_fw/fw/codecs/vp8/parser/Android.mk
index df0d270..433dca6 100644
--- a/mix_vbp/viddec_fw/fw/codecs/vp8/parser/Android.mk
+++ b/mix_vbp/viddec_fw/fw/codecs/vp8/parser/Android.mk
@@ -23,3 +23,6 @@ LOCAL_SHARED_LIBRARIES := \
         liblog
 
 include $(BUILD_SHARED_LIBRARY)
+
+PREBUILT_PROJECT := libmix
+include $(BUILD_PREBUILT_BUNDLE_CREATE)
diff --git a/mix_vbp/viddec_fw/fw/parser/Android.mk b/mix_vbp/viddec_fw/fw/parser/Android.mk
index 646e910..f3ecf18 100644
--- a/mix_vbp/viddec_fw/fw/parser/Android.mk
+++ b/mix_vbp/viddec_fw/fw/parser/Android.mk
@@ -59,3 +59,6 @@ LOCAL_CFLAGS += -DUSE_AVC_SHORT_FORMAT
 LOCAL_SRC_FILES += vbp_h264secure_parser.c
 endif
 include $(BUILD_SHARED_LIBRARY)
+
+PREBUILT_PROJECT := libmix
+include $(BUILD_PREBUILT_BUNDLE_CREATE)
diff --git a/videodecoder/Android.mk b/videodecoder/Android.mk
index 32a1851..76baa1e 100644
--- a/videodecoder/Android.mk
+++ b/videodecoder/Android.mk
@@ -65,3 +65,6 @@ LOCAL_CFLAGS += -DUSE_HW_VP8
 endif
 
 include $(BUILD_SHARED_LIBRARY)
+
+PREBUILT_PROJECT := libmix
+include $(BUILD_PREBUILT_BUNDLE_CREATE)
diff --git a/videoencoder/Android.mk b/videoencoder/Android.mk
index 480b5a1..42826fd 100644
--- a/videoencoder/Android.mk
+++ b/videoencoder/Android.mk
@@ -116,3 +116,6 @@ LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libintelmetadatabuffer
 
 include $(BUILD_SHARED_LIBRARY)
+
+PREBUILT_PROJECT := libmix
+include $(BUILD_PREBUILT_BUNDLE_CREATE)
-- 
1.7.4.1

